dropwizard-oauth/src/test/groovy/smartthings/dw/oauth/SpringSecurityAuthenticatorSpec.groovy

```diff
@@ -273,4 +275,26 @@ class SpringSecurityAuthenticatorSpec extends Specification {
         actual.get().additionalInformation == ["principal": "device:1234", "exp": 3025890541]
     }
 
+    def 'token with trailing authorization headers are cleaned correctly'() {
+        when:
+        def cleanToken = springSecurityAuthenticator.removeTrailingAuthorizationHeaders(tokenValue)
+
+        then:
+        cleanToken == result
+        noExceptionThrown()
+
+        where:
+
+        tokenValue      | result
+        token + ", Signature keyId=\"/SmartThings/33:0a:6e:17:da:b2:2c:ee:63:6c:ee:9e:62:9b:07" +
```

Comment:
This is a bit odd.

Is the goal to send both a standard Bearer token and HTTP Signature in 1 Authorization header?

Reply:
Not in the long run but in a transition period we would like to be able to move from OAuth to HTTP Signature with zero downtime. This means we need to send both headers until the endpoint supports both ways of authorization. Then the OAuth can be removed from the Authorization header.

Reply:
The dropwizard-oauth currently supports the ability to override it's default behavior with something custom.

The `SpringSecurityAuthenticator` your changes are in are defined by Guice as an Optional binding.  Meaning you can make your own authenticator and override it's injection into the auth mechanisms.

https://github.com/SmartThingsOSS/dropwizard-common/blob/master/dropwizard-oauth/src/main/java/smartthings/dw/oauth/AuthModule.java#L22

I'd suggest closing this PR, and overriding that class in your local project to do whatever custom logic your team needs.